esp-hal/src/gpio/mod.rs

```diff
@@ -1956,20 +1956,13 @@ macro_rules! gpio {
             )
         )+
     ) => {
-        #[doc(hidden)]
-        pub trait GpioExt {
-            type Parts;
-            fn split(self) -> Self::Parts;
-        }
-
         paste::paste! {
-            impl GpioExt for GPIO {
-                type Parts = Pins;
-                fn split(self) -> Self::Parts {
+            impl GPIO {
+                pub(crate) fn pins(&self) -> Pins {
```

Comment:
I'm not sure it matters or not given that this is internal, but should be consume GPIO here by using `pins(self) -> Pins`?

Reply:
Yeah wasn't sure either so I left it as a reference, can change it if you'd like but not sure how much it matters 😅

Reply:
I guess it would look cleaner if it consumes `self` - but given it's not public it's probably just a matter of taste

Reply:
Let's change this to self just to be consistent, maybe one day, years in the future, we will be saved by this choice :rofl: